### PR TITLE
Refactor/remove get_active_exchange_markets function

### PR DESCRIPTION
**Pull request by petioptrv**
**Before submitting this PR, please make sure**:

- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
This PR removes the `get_active_exchange_markets` function from the `OrderBookDataSource` class and everywhere where it is used.

It also refactors parts of the code for `liquid` and `eterbase` exchanges that relied partially on that function.

Finally, it fixes some `binance` test cases that needed to `asyncio.sleep` to complete.

**Tests performed by the developer**:
- Removes no longer relevant test cases.
- Adds tests for the affected parts of the `liquid` and `eterbase` code.

**Tips for QA testing**:
- Test that `liquid` exchange is capable of retrieving trading pairs adequately.
- Test that `eterbase` exchange is capable of retrieving trading pairs adequately.

Edit: Also, moves `MockAPIOrderBookDataSrouce` to the `test` directory to avoid having to test a test module, and because that's where it belongs upon investigating its usages.

### CODE REVIEW

File: test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py [hunk 1]
```diff
@@ -288,8 +288,10 @@ def test_create_websocket_connection_exception_raised(self, mock_ws):
         self.assertTrue(self._is_logged("NETWORK",
                                         "Unexpected error occured when connecting to WebSocket server. Error: TEST ERROR."))
 
+    @patch("hummingbot.connector.exchange.binance.binance_api_order_book_data_source"
+           ".BinanceAPIOrderBookDataSource._sleep")
```
Comment (CrimsonJacket) [inline test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py:292]:
```suggestion
patch("hummingbot.core.data_type.order_book_tracker_data_source.OrderBookTrackerDataSource._sleep")
```

File: test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py [hunk 2]
```diff
@@ -313,8 +315,10 @@ def test_listen_for_trades_exception_raised_when_connecting(self, mock_ws):
         self.assertTrue(self._is_logged("NETWORK", "Unexpected error occured when connecting to WebSocket server. Error: TEST ERROR."))
         self.assertTrue(self._is_logged("ERROR", "Unexpected error with WebSocket connection. Retrying after 30 seconds..."))
 
+    @patch("hummingbot.connector.exchange.binance.binance_api_order_book_data_source"
+           ".BinanceAPIOrderBookDataSource._sleep")
```
Comment (CrimsonJacket) [inline test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py:319]:
```suggestion
patch("hummingbot.core.data_type.order_book_tracker_data_source.OrderBookTrackerDataSource._sleep")
```

File: test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py [hunk 3]
```diff
@@ -382,8 +386,10 @@ def test_listen_for_trades_successful(self, mock_ws):
 
         self.assertTrue(12345, msg.trade_id)
 
+    @patch("hummingbot.connector.exchange.binance.binance_api_order_book_data_source"
+           ".BinanceAPIOrderBookDataSource._sleep")
```
Comment (CrimsonJacket) [inline test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py:390]:
```suggestion
patch("hummingbot.core.data_type.order_book_tracker_data_source.OrderBookTrackerDataSource._sleep")
```

File: test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py [hunk 4]
```diff
@@ -393,8 +399,10 @@ def test_listen_for_order_book_diffs_cancelled_when_connecting(self, mock_ws):
             )
             self.ev_loop.run_until_complete(self.listening_task)
 
+    @patch("hummingbot.connector.exchange.binance.binance_api_order_book_data_source"
+           ".BinanceAPIOrderBookDataSource._sleep")
```
Comment (CrimsonJacket) [inline test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py:403]:
```suggestion
patch("hummingbot.core.data_type.order_book_tracker_data_source.OrderBookTrackerDataSource._sleep")
```

**Review by CrimsonJacket — CHANGES_REQUESTED** [review]
Minor issue with patching of `_sleep` function in unit tests.

**Review by RC-13 — DISMISSED** [review]
Tested on liquid and no issue with fetching markets, skipped eterbase (deprecated). LGTM!

File: test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py (hunk 1 above)
Comment (petioptrv) [reply test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py:292]: My thinking with patching the `BinanceAPIOrderBookDataSource` method was that it won't affect other module's test cases, which was the original intent behind not directly patching `asyncio.sleep`

Comment (CrimsonJacket) [reply test/hummingbot/connector/exchange/binance/test_binance_api_order_book_data_source.py:292]: The patching of `OrderBookTrackerDataSource._sleep` does not patch the `asyncio.sleep`, 
Nonetheless, either way it works, cuz `_sleep()` is inherited from `OrderBookTrackerDataSource`.

**Comment by petioptrv** [comment]
@RC-13, latest changes were to test cases only

**Review by CrimsonJacket — APPROVED** [review]
Changes LGTM

**Pull request merged**